helpers/helpers_emba_prepare.sh

```diff
@@ -319,29 +319,53 @@ detect_root_dir_helper() {
         # remove the interpreter path from the full path:
         R_PATH="${R_PATH//$INTERPRETER_ESCAPED/}"
         ROOT_PATH+=( "$R_PATH" )
-        export RTOS=0
         MECHANISM="binary interpreter"
       done
     done
-  else
+  fi
+  if [[ ${#ROOT_PATH[@]} -eq 0 ]]; then
```

Comment:
This line is 4x in the code - maybe a little refactoring?

Reply:
would result in calling helper from helper. I know we have this multiple times but currently it is very clear and readable. If we are going to add more rules we need a redesign.